### PR TITLE
Bees are rendered docile for 30 seconds by smoke

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Bees are now made docile by smoke of any sort, including smoke
grenades, chemical reactions, nanofrost and harvester smoke.
/:cl:
- Smoke now affects mob/living, but doesn't do anything unless
  mob/living/carbon, so we can have a smoke_act() proc that bees can react
  to.
- Smoke uses timers
- Admin defined some things
- Spelling error in spellbooks

### CODE REVIEW

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 1]
```diff
@@ -51,17 +51,20 @@
...
 	if(lifetime<1)
 		return 0
-	if(C.internal != null || C.has_smoke_protection())
-		return 0
-	if(C.smoke_delay)
-		return 0
-	C.smoke_delay++
-	addtimer(src, "remove_smoke_delay", 10, FALSE, C)
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
```
Comment (RemieRichards) [inline code/game/objects/effects/effect_system/effects_smoke.dm]: shouldn't this all just be in ```/carbon/smoke_act()```?

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 2]
```diff
@@ -181,6 +186,10 @@
 	if(beehome)
 		beehome.bee_resources = min(beehome.bee_resources + growth, 100)
 
+/mob/living/simple_animal/hostile/poison/bees/proc/tray_cooldown(obj/machinery/hydroponics/H)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: why is this a proc on the bee and not the tray? this should be a proc on the tray

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 3]
```diff
@@ -286,5 +295,22 @@
...
+		AIStatus = AI_OFF
+		stop_automated_movement = TRUE
+		SpinAnimation(speed=100 + rand(-20, 20), clockwise=prob(50))
+		name = "docile [name]"
+	addtimer(src, "smoke_timeout", BEE_SMOKE_SLEEP)
+
+/mob/living/simple_animal/hostile/poison/bees/proc/smoke_timeout()
+	smoke_counter--
+	if(!smoke_counter)
+		AIStatus = initial(AIStatus)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: this should always really be ```AI_IDLE```

File: code/game/objects/effects/effect_system/effects_smoke.dm (hunk 1 above)
Comment (coiax) [inline code/game/objects/effects/effect_system/effects_smoke.dm]: good point

**Comment by bobdobbington** [comment]
30 seconds is way, way too long for something that regularly just belches out of vents as a random event. 15 or even 20 would be fine.

**Comment by hornygranny** [comment]
this makes metastation chaplain the bee god

**Comment by Kromgar** [comment]
Bee's don't smoke!

**Comment by KorPhaeron** [comment]
Someone should fix up this PR or I'll reopen it when coiax is back

**Pull request closed without merging**